packages/wallet-stamper/src/stamper.ts

```diff
@@ -27,16 +32,37 @@ export class WalletStamper implements TStamper {
 
     // Determine the signature scheme based on the wallet type.
     const scheme =
-      this.wallet.type === WALLET_TYPE_SOLANA
+      this.wallet.type === WalletType.Solana
         ? SIGNATURE_SCHEME_TK_API_ED25519
-        : SIGNATURE_SCHEME_TK_API_SECP256K1;
+        : SIGNATURE_SCHEME_TK_API_SECP256K1_EIP191;
 
     let publicKey: string;
     try {
-      publicKey =
-        this.wallet.type === WALLET_TYPE_SOLANA
-          ? this.wallet.recoverPublicKey()
-          : await this.wallet.recoverPublicKey(payload, signature);
+      // For Ethereum, we need to recover the public key from the signature over the payload.
+      // This is because recovering the SECP256K1 public key requires a signed message.
+      // This avoids the an additional call to the wallet to get the public key.
```

Comment:
tiny nit: let's choose one of `the / an`

Reply:
can you explain how we avoid the additional call? wouldn't we need an extra one regardless in order to get that public key?

Reply:
With solana I can call `getPublicKey` without triggering a signature request but with ethereum we need to derive it from the signature, so instead of calling `getPublicKey` we derive it from the signature over the payload

Reply:
OH! right right, that's beautiful!